Code/GraphMol/RGroupDecomposition/RGroupCore.cpp

```diff
@@ -54,6 +64,197 @@ void RCore::findIndicesWithRLabel() {
   }
 }
 
+RWMOL_SPTR RCore::extractCoreFromMolMatch(
+    bool &hasCoreDummies, const ROMol &mol, const MatchVectType &match,
+    const RGroupDecompositionParameters &params) const {
+  auto extractedCore = boost::make_shared<RWMol>(mol);
+  std::set<int> atomIndicesToKeep;
+  std::vector<Bond *> newBonds;
+  std::map<Atom *, int> dummyAtomMap;
+  for (auto &pair : match) {
+    const auto queryAtom = core->getAtomWithIdx(pair.first);
+    auto const targetAtom = extractedCore->getAtomWithIdx(pair.second);
+    if (int rLabel; queryAtom->getPropIfPresent(RLABEL, rLabel)) {
+      targetAtom->setProp(RLABEL, rLabel);
+    }
+    if (int rLabelType; queryAtom->getPropIfPresent(RLABEL_TYPE, rLabelType)) {
+      targetAtom->setProp(RLABEL_TYPE, rLabelType);
+    }
+
+    if (queryAtom->getAtomicNum() == 0) {
```

Comment:
Maybe if(!hasCoreDummies && queryAtom->getAtomicNum() == 0) for a possibly useless optimization.

Reply:
> Maybe if(!hasCoreDummies && queryAtom->getAtomicNum() == 0) for a possibly useless optimization.

I'll change that.